arctic/chunkstore/chunkstore.py

```diff
@@ -126,6 +126,44 @@ def list_symbols(self):
     def _get_symbol_info(self, symbol):
         return self._symbols.find_one({SYMBOL: symbol})
 
+    def rename(self, from_symbol, to_symbol):
+        """
+        Rename a symbol
+
+        Parameters
+        ----------
+        from_symbol: str
+            the existing symbol that will be renamed
+        to_symbol: str
+            the new symbol name
+        """
+
+        sym = self._get_symbol_info(from_symbol)
+        if not sym:
+            raise NoDataFoundException('No data found for %s' % (from_symbol))
+
+        if self._get_symbol_info(to_symbol) is not None:
+            raise Exception('Symbol %s already exists' % (to_symbol))
+
+        chunks = []
+        for x in self._collection.find({SYMBOL: from_symbol}, sort=[(START, pymongo.ASCENDING)],):
+            chunks.append(x)
+
+        bulk = self._collection.initialize_unordered_bulk_op()
+        for chunk in chunks:
+            chunk.pop('_id', None)
+            chunk[SYMBOL] = to_symbol
+            chunk[SHA] = self._checksum(to_symbol, chunk)
```

Comment:
Could be optimised to remove the symbol from the SHA, then this turns into a simple `update` operation.